### PR TITLE
feat: add advanced schedule settings

### LINKED ISSUE #6047: scheduling: advanced scheduling controls
Currently, workspace schedules are only a recommendation and there is no way for admins to enforce a particular shutdown behavior on workspaces. #3630, #5600.

We have seen a decent amount of feedback that warrants us supporting both "max" and "default" TTL as they serve different use cases. For example, max TTL to enforce security patches, but default TTL to ensure workspaces quickly shut down when inactive. 

Let's re-introduce the "max TTL" while also keeping the "default TTL" behavior for workspaces. This will allow workspaces to have the following behavior:

- Workspaces will auto-shutdown after 2 hours of inactivity (default TTL), but cannot be alive for over 72 hours (max TTL).

Also, some admins may not want users to change the default TTL behavior of their workspaces. For example,

- All workspaces on `template x` will auto shut down after 2 hours of inactivity. Users cannot change this behavior.

To summarize:

- [x] Introduce max lifetime value in template settings (while keeping default TTL): #6496
- [ ] Add a toggle in template settings to disable user-controlled shutdown behavior: #6497

### LINKED ISSUE #6497: scheduling: Add a toggle in template settings to disable user-controlled shutdown behavior
Contributes to #6047

### PR DESCRIPTION
**Pull request by rodrimaia**
image

Closes https://github.com/coder/coder/issues/6047
Closes https://github.com/coder/coder/issues/6497

### CODE REVIEW

**Review by BrunoQuaresma — APPROVED** [review]
LGTM

**Review by deansheather — COMMENTED** [review]
Looks good, but can we change `auto-start` => `autostart` and `auto-stop` to `autostop`? Sorry if my original diagrams were incorrect but this is what we've settled on now

File: site/src/pages/CreateTemplatePage/CreateTemplateForm.tsx [hunk 1]
```diff
@@ -341,6 +343,55 @@ export const CreateTemplateForm: FC<CreateTemplateFormProps> = ({
               type="number"
             />
           </Stack>
+          <Stack direction="column">
+            <Stack direction="row" alignItems="center">
+              <Checkbox
+                id="allow_user_autostart"
+                size="small"
+                color="primary"
+                disabled={isSubmitting || !canSetMaxTTL}
```
[the remark this answers is not in the document]
Comment (rodrimaia) [reply site/src/pages/CreateTemplatePage/CreateTemplateForm.tsx]: fixed

**Comment by rodrimaia** [comment]
> @rodrimaia is just a minor thing. The secondary text of the new checkbox does not match the current style. You can see an existing one with a smaller secondary text and a more gray color.
> 
> Screen Shot 2023-04-12 at 10 12 13

Fixed! thank you for noticing, Bruno! :)

**Pull request merged**